docs/overview.adoc

```diff
@@ -19,25 +19,140 @@ For example using Kroxylicious proxy it's easier to experiment with alternative
 
 == How it works
 
-// TODO include a diagram
-
-When a Kafka protocol client (typically a producer, consumer or admin client) connects to a proxy instance it is first handled by a pluggable _network filter_
-This handles things like:
-
-* SSL
-* Possibly initial client requests such as `ApiVersions`, `SaslHandshake` and `SaslAuthenticate`
-
-The network filter then
-
-. determines which broker to connect the client to
-. instantiates a chain of protocol filters for processing all further messages.
-. removes itself from the further processing
-
-A _filter chain_ consists of one or more pluggable _protocol filters_.
+First let's define the concepts in the landscape surrounding Kroylicious.
+
+. _Kafka Client_, or _Client_ refers to any client application using a Kafka Client library to talk to a *Kafka Cluster*.
+. _Kafka Cluster_ or _Cluster_ refers to a cluster comprising one or more Kafka Brokers.
+. _Downstream_ refers to the area between Kafka Client and Kroxylicious.
+. _Upstream_ refers to the area between Kroxylicious and a Kafka Cluster.
+
+[a2s, format="svg"]
```

Comment:
I used https://github.com/asciitosvg/asciitosvg here just to get something done quickly.   I'd be happy to change to another format.